cmd/drone-server/server.go

```diff
@@ -175,6 +175,24 @@ var flags = []cli.Flag{
 		Usage:  "token to secure prometheus metrics endpoint",
 		Value:  "",
 	},
+	cli.StringFlag{
+		EnvVar: "DRONE_VAULT_AUTH_TYPE",
+		Name:   "drone-vault-auth-type",
+		Usage:  "auth backend type used for connecting to vault",
+		Value:  "token",
```

Comment:
envisioning a case where we'd switch on this auth type, "token" as the current default, "kubernetes" for this particular auth method, and any others from https://www.vaultproject.io/docs/auth/index.html in the future